openhantek/src/widgets/dsowidget.cpp

```diff
@@ -487,3 +381,125 @@ void DsoWidget::applyColors() {
     markerFrequencybaseLabel->setPalette(palette);
     markerInfoLabel->setPalette(palette);
 }
+
+ChannelWidgets::ChannelWidgets(Settings::Channel *channel, Settings::View *view, const Dso::ModelSpec *spec, QWidget *parent)
+        : QWidget(parent), m_view(view), channel(channel), m_spec(spec) {
+    QPalette tablePalette = palette();
+    ChannelID channelId = channel->channelID();
+
+    tablePalette.setColor(QPalette::WindowText, m_view->screen.voltage(channelId));
+    measurementNameLabel->setText(channel->name());
+    measurementNameLabel->setPalette(tablePalette);
+    measurementMiscLabel->setPalette(tablePalette);
+    measurementGainLabel->setPalette(tablePalette);
+    measurementGainLabel->setAlignment(Qt::AlignRight);
+
+    tablePalette.setColor(QPalette::WindowText, m_view->screen.spectrum(channelId));
+    measurementMagnitudeLabel->setPalette(tablePalette);
+    measurementMagnitudeLabel->setAlignment(Qt::AlignRight);
+    measurementAmplitudeLabel->setPalette(tablePalette);
+    measurementAmplitudeLabel->setAlignment(Qt::AlignRight);
+    measurementFrequencyLabel->setPalette(tablePalette);
+    measurementFrequencyLabel->setAlignment(Qt::AlignRight);
+
+    layout->setMargin(0);
+
+    layout->setColumnMinimumWidth(0, 64);
+    layout->setColumnMinimumWidth(1, 32);
+    layout->setColumnStretch(2, 2);
+    layout->setColumnStretch(3, 2);
+    layout->setColumnStretch(4, 3);
+    layout->setColumnStretch(5, 3);
+
+    layout->addWidget(measurementNameLabel, 0, 0);
+    layout->addWidget(measurementMiscLabel, 0, 1);
+    layout->addWidget(measurementGainLabel, 0, 2);
+    layout->addWidget(measurementMagnitudeLabel, 0, 3);
+    layout->addWidget(measurementAmplitudeLabel, 0, 4);
+    layout->addWidget(measurementFrequencyLabel, 0, 5);
+
+    setMeasurementVisible();
+
+    if (!channel->isMathChannel()) {
+        updateVoltageCoupling();
+        connect(channel->voltage(), &Dso::Channel::couplingIndexChanged, this, &ChannelWidgets::updateVoltageCoupling);
+    } else {
+        updateMathMode();
+        connect((Settings::MathChannel *)channel, &Settings::MathChannel::mathModeChanged, this, &ChannelWidgets::updateMathMode);
+    }
+    updateVoltageUsed();
+    updateSpectrumDetails();
+
+    connect(channel, &Settings::Channel::gainChanged, this, &ChannelWidgets::updateVoltageDetails);
+    connect(channel, &Settings::Channel::visibleChanged, this, &ChannelWidgets::updateVoltageUsed);
+
+    connect(channel->spectrum(), &Settings::Spectrum::magnitudeChanged, this, &ChannelWidgets::updateSpectrumDetails);
+    connect(channel->spectrum(), &Settings::Spectrum::visibleChanged, this, &ChannelWidgets::updateSpectrumDetails);
+}
+
+/// \brief Show/Hide a line of the measurement table.
+void ChannelWidgets::setMeasurementVisible() {
```

Comment:
This should be renamed to `updateMeasurementVisibility` as we do not *set* anything without any arguments.